lua/calltree/ui.lua

```diff
@@ -345,11 +345,15 @@ end
 --
 -- keep_open : bool - if true, and the panel is open,
 -- the panel will be left open when this function terminates.
+--
+-- returns bool | err: true when panel was opened by this call,
+-- false if it was closed. err is not nil in case panel could
+-- not be opened.
 M.toggle_panel = function(keep_open)
     local ctx = ui_req_ctx()
     if ctx.state == nil then
         notify.notify_popup_with_timeout("Cannot toggle panel until LSP method is called.", 1750)
-        return
+        return false, "Cannot toggle panel until LSP method is called."
```

Comment:
cool, so you just want to return this so some calling function can check if the panel is open programatically ?

Reply:
Exactly. As of right now I use it like this
```
local _, err = require("calltree.ui").toggle_panel()
if err ~= nil then
	vim.lsp.buf.document_symbol()
end
```

Reply:
Sounds smart.